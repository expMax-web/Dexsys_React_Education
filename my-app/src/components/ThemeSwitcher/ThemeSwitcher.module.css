.checkboxContainer {
  position: absolute;
  top: 25px;
  right: 50px;
}
.checkbox {
  position: absolute;
  z-index: -1;
  opacity: 0;
  margin: 10px 0 0 20px;
}
.checkbox + label {
  position: relative;
  padding: 0 0 0 60px;
  cursor: pointer;
}
.checkbox + label:before {
  content: "";
  position: absolute;
  top: -4px;
  left: 0;
  width: 50px;
  height: 26px;
  border-radius: 13px;
  background: #2b2b2c;
  box-shadow: inset 0 2px 3px rgba(0, 0, 0, 0.2);
  transition: 0.2s;
}
.checkbox + label:after {
  content: "";
  position: absolute;
  top: -2px;
  left: 2px;
  width: 22px;
  height: 22px;
  border-radius: 10px;
  background: #fff;
  box-shadow: 0 2px 5px rgba(0, 0, 0, 0.3);
  transition: 0.2s;
}
.checkbox:checked + label:before {
  background: #c6d9df;
}
.checkbox:checked + label:after {
  left: 26px;
}
.checkbox:focus + label:before {
  box-shadow: inset 0 2px 3px rgba(0, 0, 0, 0.2),
    0 0 0 3px rgba(255, 255, 0, 0.7);
}

.labelDark {
  color: white;
}
