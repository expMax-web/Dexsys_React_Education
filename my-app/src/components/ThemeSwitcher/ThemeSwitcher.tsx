import React, { ChangeEvent, useState } from "react";
import cn from "classnames";

import { useDarkTheme } from "../../hooks/useDarkTheme";

import styles from "./ThemeSwitcher.module.scss";

export const ThemeSwitcher: React.FC = () => {
  const { isDark, setIsDark } = useDarkTheme();
  const [pointer, setPointer] = useState("enable dark theme");

  const toggleTheme = (event: ChangeEvent<HTMLInputElement>) => {
    setIsDark(event.target.checked);
    if (isDark) setPointer("enable dark theme");
    if (!isDark) setPointer("turn off dark theme");
  };

  return (
    <div className={styles.CheckboxContainer}>
      <input
        type="checkbox"
        className={styles.Checkbox}
        id="checkbox"
        onChange={toggleTheme}
        checked={isDark}
      />
      <label
        htmlFor="checkbox"
        className={cn({
          [styles.LabelDark]: isDark,
        })}
      >
        {pointer}
      </label>
    </div>
  );
};
