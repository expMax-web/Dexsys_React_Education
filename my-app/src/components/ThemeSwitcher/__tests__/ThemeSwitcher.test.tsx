import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import { ThemeSwitcher } from "../ThemeSwitcher";

describe("ThemeSwitcher test suite", () => {
  test("should be render correctly ", () => {
    const { container } = render(<ThemeSwitcher />);
    expect(container).toMatchSnapshot();
  });
  test("Use dark theme correctly", () => {
    const { container } = render(<ThemeSwitcher />);

    userEvent.click(screen.getByRole("checkbox"));
    expect(container).toHaveTextContent("turn off dark theme");
  });
  test("Use dark theme correctly on label click", () => {
    const { container } = render(<ThemeSwitcher />);

    userEvent.click(screen.getByText("enable dark theme"));

    expect(container).toHaveTextContent("turn off dark theme");
  });
});
