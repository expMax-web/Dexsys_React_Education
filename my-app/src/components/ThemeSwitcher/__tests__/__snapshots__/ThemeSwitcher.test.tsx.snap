// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmitButton test suite should be render correctly  1`] = `
<div>
  <div
    class="CheckboxContainer"
  >
    <input
      class="Checkbox"
      id="checkbox"
      type="checkbox"
    />
    <label
      class=""
      for="checkbox"
    >
      enable dark theme
    </label>
  </div>
</div>
`;

exports[`ThemeSwitcher test suite should be render correctly  1`] = `
<div>
  <div
    class="CheckboxContainer"
  >
    <input
      class="Checkbox"
      id="checkbox"
      type="checkbox"
    />
    <label
      class=""
      for="checkbox"
    >
      enable dark theme
    </label>
  </div>
</div>
`;
